src/fem/fe_system.cc

```diff
@@ -0,0 +1,416 @@
+  {
+    this->node_matrix.submat(offset, offset, offset+fe_[i]->n_dofs()-1, offset+fe_[i]->n_dofs()-1)
+      = fe_[i]->node_matrix;
+      
+    offset += fe_[i]->n_dofs();
+  }
+}
+
+
+template<unsigned int dim, unsigned int spacedim>
+FEInternalData *FESystem<dim,spacedim>::initialize(const Quadrature<dim> &q, UpdateFlags flags)
+{
+  FEInternalData *data = new FEInternalData;
+  std::vector<FEInternalData *> fe_data;
+  
+  // first initialize the base FE
+  if ((flags & update_values) || (flags & update_gradients))
+    for (auto fe : fe_)
+      fe_data.push_back(fe->initialize(q, flags));
+
+  if (flags & update_values)
+  {
+    // fill values of basis functions
+    data->basis_vectors.resize(q.size(), std::vector<arma::vec>(number_of_dofs, arma::vec(n_components_)));
+    
+    unsigned int comp_offset = 0;
+    unsigned int dof_offset = 0;
+    for (unsigned int f=0; f<fe_.size(); f++)
+    {
+      if (fe_[f]->n_components() == 1)
+      {
+        // for scalar base FE copy only one value per point and dof
+        for (unsigned int i=0; i<q.size(); i++)
+          for (unsigned int n=0; n<fe_[f]->n_dofs(); n++)
+            data->basis_vectors[i][dof_offset+n][comp_offset] = fe_data[f]->basis_values[i][n];
+      }
+      else
+      {
+        // for vector-valued base FE copy the values to subvector
+        for (unsigned int i=0; i<q.size(); i++)
+          for (unsigned int n=0; n<fe_[f]->n_dofs(); n++)
+            data->basis_vectors[i][dof_offset+n].subvec(comp_offset,comp_offset+fe_[f]->n_components()-1) = fe_data[f]->basis_vectors[i][n];
+      }
+      comp_offset += fe_[f]->n_components();
+      dof_offset += fe_[f]->n_dofs();
+    }
+  }
+
+  if (flags & update_gradients)
+  {
+    // fill gradients of basis functions
+    data->basis_grad_vectors.resize(q.size(), std::vector<arma::mat>(number_of_dofs, arma::mat(n_components_,dim)));
+    
+    unsigned int comp_offset = 0;
+    unsigned int dof_offset = 0;
+    for (unsigned int f=0; f<fe_.size(); f++)
+    {
+      if (fe_[f]->n_components() == 1)
+      {
+        // for scalar base FE copy values to one row per point and dof
+        for (unsigned int i=0; i<q.size(); i++)
+          for (unsigned int n=0; n<fe_[f]->n_dofs(); n++)
+            data->basis_grad_vectors[i][dof_offset+n].row(comp_offset) = fe_data[f]->basis_grads[i].row(n);
+      }
+      else
+      {
+        // for vector-valued base FE copy the values to submatrix
+        for (unsigned int i=0; i<q.size(); i++)
+          for (unsigned int n=0; n<fe_[f]->n_dofs(); n++)
+            data->basis_grad_vectors[i][dof_offset+n].submat(comp_offset,0,comp_offset+fe_[f]->n_components()-1,dim-1) = fe_data[f]->basis_grad_vectors[i][n];
+      }
+      comp_offset += fe_[f]->n_components();
+      dof_offset += fe_[f]->n_dofs();
+    }
+  }
+  
+  for (auto d : fe_data) delete d;
+
+  return data;
+}
+
+template<unsigned int dim, unsigned int spacedim> inline
+void FESystem<dim,spacedim>::fill_fe_values(
```

Comment:
What have to be computed on every cell? The implementation in FiniteElement just copy values on reference element.
For RT, we should just apply mapping to values on reference element.